src/worker/rpc.rs

```diff
@@ -197,13 +202,47 @@ pub async fn run_worker(
                 unreachable!()
             }
             _ = heartbeat => { unreachable!() }
+            _ = hw_state_updater => { unreachable!() }
         }
     }))
 }
 
 const MAX_RUNNING_DOWNLOADS: usize = 32;
 const MAX_ATTEMPTS: u32 = 8;
 
+async fn update_hw_state(
+    state_ref: WorkerStateRef,
+    hw_poll_interval: Duration,
+    mut hw_sampler: Result<HwSampler, psutil::Error>,
+) {
+    let mut poll_interval = tokio::time::interval(hw_poll_interval);
+    loop {
+        poll_interval.tick().await;
+        match &mut hw_sampler {
+            Ok(sampler) => {
+                let hw_state = sampler.fetch_hw_state();
+                match hw_state {
+                    Ok(new_state) => {
+                        state_ref.get_mut().hardware_state = new_state;
+                    }
+                    Err(error) => {
+                        state_ref
```

Comment:
Why is it cleared here?

Reply:
It was my requirement. It is a kind of sanity check, that we are not simply copying a last state when monitoring fails

Reply:
Hmm, but why is only the CPU state cleared then? The HW states have a timestamp, we should instead check if the received timestamp is newer to recognize potential errors. The overview request could also ask for X records since timestamp T, otherwise if tako buffers records and sends more of them, it will either needlessly resend already sent records, or it would have to delete the buffer after every overview call.